src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -136,7 +139,28 @@ private void updateJobs(Map<String, String> oldBaseUrls) {
                                 }
                             }
                         });
+                Jenkins.get().getAllItems(MultiBranchProject.class)
+                        .forEach(project -> processMultibranchProject(project, changedServerIds));
             }
         }
     }
+
+    @SuppressWarnings("unchecked")
+    private void processMultibranchProject(MultiBranchProject project, Set<String> changedServerIds) {
+        List<SCMSource> sources = (List<SCMSource>) project.getSources().stream().map(scm -> {
+            if (scm instanceof BitbucketSCMSource) {
+                BitbucketSCMSource src = (BitbucketSCMSource) scm;
+                if (changedServerIds.contains(src.getServerId())) {
+                    return new BitbucketSCMSource(src);
+                }
+            }
+            return scm;
+        }).collect(Collectors.toList());
+        try {
+            project.setSourcesList(sources);
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, String.format("Error updating configuration for MultibranchProject %s.",
+                    project.getName()), e);
```

Comment:
Do we need to also propagate the exception here, e.g. as some type of runtime exception?

Reply:
I don't think so- a log entry is the same way we handle errors updating ParameterizedJobs so the approach is consistent. An unchecked exception would stop the update process entirely, which affects other projects and other plugins so I think it's best to leave this as is.